### PR TITLE
Leaf909 data importer callbacks

### PR DESCRIPTION
**Pull request by jeanmarcablevets**
Updated JS API module to be asynchronous and changed importer accordingly.

### CODE REVIEW

File: LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl [hunk 1]
```diff
@@ -139,7 +140,7 @@
 
     function checkFormatExisting(column) {
         for (var i = 1; i < sheet_data.cells.length; i++) {
-            var value = sheet_data.cells[i][column].toString();
+            var value = sheet_data.cells[i] !== undefined && sheet_data.cells[i][column] !== undefined ? sheet_data.cells[i][column].toString() : '';
```
Comment (thomasnguyen95) [inline LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl]: I personally have a preference for ```typeof sheet_data.cells[i] !== "undefined"``` which I feel is safer, but I don't believe this needs to be changed.

Comment (jeanmarcablevets) [reply LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl]: Addressed in commit `bd5`

File: LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl [hunk 2]
```diff
@@ -332,29 +335,29 @@
...
 
-                    // recordID is the recordID of the newly created request, it's 0 if there was an error
+                    /* recordID is the recordID of the newly created request, it's 0 if there was an error */
                     if (recordID > 0) {
                         createdRequests++;
                         requestStatus.html(createdRequests + ' out of ' + (sheet_data.cells.length - 1) + ' requests completed, ' + failedRequests.length + ' failures.');
 
-                        //if (changeToInitiator !== undefined && changeToInitiator != null) {
-                        // set the initiator so they can see the request associated with their availability
+                        /*if (changeToInitiator !== undefined && changeToInitiator != null) { */
```
Comment (thomasnguyen95) [inline LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl]: Might want to delete this commented code

File: LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl [hunk 3]
```diff
@@ -332,29 +335,29 @@
...
+                        /* set the initiator so they can see the request associated with their availability */
                         portalAPI.Forms.setInitiator(
                             recordID,
                             initiator,
                             function (results) {},
                             function (err) {
                                 console.log(err);
                             });
-                        //}
+                        /*} */
```
Comment (thomasnguyen95) [inline LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl]: Might want to delete this commented code

Comment (jeanmarcablevets) [reply LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl]: Addressed in commit `7c5`

File: LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl (hunk 2 above)
Comment (jeanmarcablevets) [reply LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl]: Addressed in commit `7c5`

**Comment by jeanmarcablevets** [comment]
> Ran it on a spreadsheet with 1120 rows and got this error in the console. `Uncaught RangeError: Maximum call stack size exceeded`

@jeremyrl Spreadsheet limit should be fixed now in commit `109`

**Pull request merged**